DataFormats/TrackReco/interface/HitPattern.h

```diff
@@ -262,9 +262,9 @@ class HitPattern
     void printHitPattern(HitCategory category, int position, std::ostream &stream) const;
     void print(HitCategory category, std::ostream &stream = std::cout) const;
 
-    bool hasValidHitInFirstPixelBarrel() const; // has valid hit in PXB layer 1
-    bool hasValidHitInFirstPixelEndcap() const; // has valid hit in PXF layer 1
-
+    // has valid hit in PXB/PXF layer x 
+    bool hasValidHitInPixelLayer(bool inBarrel, uint16_t layer) const;
+    
```

Comment:
You are right. I will use the enum in https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DataFormats/SiPixelDetId/interface/PixelSubdetector.h.